File: back_bateau/src/main/java/com/batean/back_bateau/model/ShipFilters.java
package com.batean.back_bateau.model;

public class ShipFilters {
    private Integer power;
    private String marketPlace;
    private Integer nbPeopleMax;

    public ShipFilters(Integer power, String marketPlace, Integer nbPeopleMax) {
        this.power = power;
        this.marketPlace = marketPlace;
        this.nbPeopleMax = nbPeopleMax;
    }

    public int getPower() {
        return power;
    }

    public void setPower(int power) {
        this.power = power;
    }

    public String getMarketPlace() {
        return marketPlace;
    }

    public void setMarketPlace(String marketPlace) {
        this.marketPlace = marketPlace;
    }

    public int getNbPeopleMax() {
        return nbPeopleMax;
    }

    public void setNbPeopleMax(int nbPeopleMax) {
        this.nbPeopleMax = nbPeopleMax;
    }

    public boolean isNull() {
        return !hasPower() && !hasMarketPlace() && !hasNbPeopleMax();
    }

    public boolean hasPower() {
        return !(power == null || power == 0);
    }

    public boolean hasMarketPlace() {
        return !(marketPlace == null || marketPlace.trim().isEmpty());
    }

    public boolean hasNbPeopleMax() {
        return!(nbPeopleMax == null || nbPeopleMax == 0);
    }
}

File: bateau_market/lib/features/ship/models/ship_filters.dart
class ShipFilters {
  final int? power;
  final String? marketPlace;
  final int? nbPeopleMax;

  ShipFilters({this.power, this.marketPlace, this.nbPeopleMax});

  ShipFilters copyWith({int? power, String? marketPlace, int? nbPeopleMax}) {
    return ShipFilters(
      power: power ?? this.power,
      marketPlace: marketPlace ?? this.marketPlace,
      nbPeopleMax: nbPeopleMax ?? this.nbPeopleMax,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (power != null) 'power': power,
      if (marketPlace != null) 'marketPlace': marketPlace,
      if (nbPeopleMax != null) 'nbPeopleMax': nbPeopleMax,
    };
  }
}
